src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INIT_SIZE = 10;
+    private static Object[] array = new Object[INIT_SIZE];
+    private int size = 0;
+    
     @Override
     public void add(T value) {
-
+        if (size == array.length - 1) {
+            increaseArray(array.length + array.length / 2 + 1);
+        }
+        array[size++] = value;
     }
-
+    
     @Override
     public void add(T value, int index) {
-
+        if (size != index) {
+            checkIndex(index);
+        }
+        if (size == array.length - 1) {
+            increaseArray(array.length + array.length / 2 + 1);
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
-
+    
     @Override
     public void addAll(List<T> list) {
-
+        Object[] a = new Object[list.size()];
+        for (int i = 0; i < a.length; i++) {
+            a[i] = list.get(i);
+        }
+        if (list.size() > array.length - size) {
+            increaseArray(list.size() + size);
+        }
+        
+        System.arraycopy(a, 0, array, size, list.size());
+        size = list.size() + size;
     }
-
+    
     @Override
     public T get(int index) {
+        checkIndex(index);
+        for (int i = 0; i < array.length; i++) {
+            if (index == i) {
+                return (T) array[i];
+            }
+        }
         return null;
```

Comment:
why do you need to iterate?

Reply:
I just was looking for the same index) 
now it is corrected.